### PR TITLE
Allow building with GHC 9.2

### PR DESCRIPTION
**Pull request by TeofilC**
Before submitting your PR, check that you've:

- [X] Bumped the version number

After submitting your PR:

- [x] Update the Changelog.md file with a link to your PR
- [ ] Check that CI passes (or if it fails, for reasons unrelated to your change, like CI timeouts)

I haven't added a new CI test for 9.2 as one of the dependencies (hjsmin) is missing from stackage nightly

### CODE REVIEW

File: yesod-bin/AddHandler.hs [hunk 1]
```diff
@@ -247,4 +250,8 @@ getSrcDir cabal = do
 #endif
     let buildInfo = allBuildInfo pd
         srcDirs = concatMap hsSourceDirs buildInfo
-    return $ fromMaybe "." $ listToMaybe srcDirs
+#if MIN_VERSION_Cabal(3, 6, 0)
+    return $ maybe "." getSymbolicPath $ listToMaybe srcDirs
+#else
+    return $ fromMaybe "." $ listToMaybe srcDirsFilePath
```
Comment (schoettl) [inline yesod-bin/AddHandler.hs:256]: I'm not a regular haskell code reviewer, so sorry if I'm wrong. But where does `srcDirsFilePath` come from? I can't find the definition in this file.

File: yesod-core/src/Yesod/Routes/TH/RenderRoute.hs [hunk 2]
```diff
@@ -182,3 +182,10 @@ notStrict = Bang NoSourceUnpackedness NoSourceStrictness
 
 instanceD :: Cxt -> Type -> [Dec] -> Dec
 instanceD = InstanceD Nothing
+
+conPCompat :: Name -> [Pat] -> Pat
+conPCompat n pats = ConP n
+#if MIN_VERSION_template_haskell(2,18,0)
+                         []
+#endif
+                         pats
```
Comment (schoettl) [inline yesod-core/src/Yesod/Routes/TH/RenderRoute.hs:191]: Isn't here an `#else` missing? Or does `ConP` actually have a different number of args in different haskell versions?

File: yesod-bin/AddHandler.hs (hunk 1 above)
Comment (TeofilC) [reply yesod-bin/AddHandler.hs]: Good spot! That was a typo the "FilePath" shouldn't be there. I've force pushed a fix

File: yesod-core/src/Yesod/Routes/TH/RenderRoute.hs (hunk 2 above)
Comment (TeofilC) [reply yesod-core/src/Yesod/Routes/TH/RenderRoute.hs:191]: Yes that's right. There's a new argument in template-haskell 2.18

**Review by snoyberg — APPROVED** [review]

**Pull request merged**